File: OrnaTrack/Stock/OrganiseStock.vb
Imports Newtonsoft.Json
Imports Newtonsoft.Json.Linq

Public Class OrganiseStock

    Dim _width As Integer = -1
    Dim _height As Integer = -1

    Private Sub OrganiseStock_Load(sender As Object, e As EventArgs) Handles MyBase.Load

    End Sub

    Private Sub MakePlace()
        FlowLayoutPanel1.Controls.Clear()
        Integer.TryParse(TrayWidthTB.Text, _width)
        Integer.TryParse(TrayHeightTB.Text, _height)

        For i = 0 To _width * _height - 1
            Dim _cntrl As New ItemsUserControl
            FlowLayoutPanel1.Controls.Add(_cntrl)
        Next

    End Sub

    Private Sub TextBox1_TextChanged(sender As Object, e As EventArgs) Handles TrayWidthTB.TextChanged
        MakePlace()
    End Sub

    Private Sub TrayHeightTB_TextChanged(sender As Object, e As EventArgs) Handles TrayHeightTB.TextChanged
        MakePlace()
    End Sub

    Private Sub Button1_Click(sender As Object, e As EventArgs) Handles Button1.Click
        Dim dr As OleDb.OleDbDataReader = DataReader("Select * From Stock_Data")

        Dim _list As New List(Of Double)
        Dim _count As Integer = 0
        While dr.Read
            _count += 1
            Dim _str As String = dr("Item_Data").ToString
            Dim _id As Integer = dr("ID")
            Dim jsonObject As JObject = JsonConvert.DeserializeObject(Of JObject)(_str)

            Try
                Dim _weight As Double = jsonObject("12").ToObject(Of Double)()
                _list.Add(_weight)
            Catch ex As Exception
            End Try

        End While



        Dim sum As Double = _list.Sum()
        Dim g = ""

    End Sub
End Class

File: FirstApp/Stock/OrganiseStock.cs
using System;
using System.Collections.Generic;
using System.Linq;
using Microsoft.VisualBasic.CompilerServices;
using Newtonsoft.Json;
using Newtonsoft.Json.Linq;

namespace FirstApp
{

    public partial class OrganiseStock
    {

        private int _width = -1;
        private int _height = -1;

        public OrganiseStock()
        {
            InitializeComponent();
        }

        private void OrganiseStock_Load(object sender, EventArgs e)
        {

        }

        private void MakePlace()
        {
            FlowLayoutPanel1.Controls.Clear();
            int.TryParse(TrayWidthTB.Text, out _width);
            int.TryParse(TrayHeightTB.Text, out _height);

            for (int i = 0, loopTo = _width * _height - 1; i <= loopTo; i++)
            {
                var _cntrl = new ItemsUserControl();
                FlowLayoutPanel1.Controls.Add(_cntrl);
            }

        }

        private void TextBox1_TextChanged(object sender, EventArgs e)
        {
            MakePlace();
        }

        private void TrayHeightTB_TextChanged(object sender, EventArgs e)
        {
            MakePlace();
        }

        private void Button1_Click(object sender, EventArgs e)
        {
            System.Data.OleDb.OleDbDataReader dr = (System.Data.OleDb.OleDbDataReader)ProjectModule.DataReader("Select * From Stock_Data");

            var _list = new List<double>();
            int _count = 0;
            while (dr.Read())
            {
                _count += 1;
                string _str = dr["Item_Data"].ToString();
                int _id = Conversions.ToInteger(dr["ID"]);
                var jsonObject = JsonConvert.DeserializeObject<JObject>(_str);

                try
                {
                    double _weight = jsonObject["12"].ToObject<double>();
                    _list.Add(_weight);
                }
                catch (Exception ex)
                {
                }

            }



            double sum = _list.Sum();
            string g = "";

        }
    }
}